ta/pkcs11/include/pkcs11_ta.h

```diff
@@ -1181,6 +1181,27 @@ enum pkcs11_key_type {
 	PKCS11_CKK_UNDEFINED_ID			= PKCS11_UNDEFINED_ID,
 };
 
+/*
+ * Valid values for attribute PKCS11_CKA_CERTIFICATE_TYPE
+ */
+enum pkcs11_certificate_type {
+	PKCS11_CKC_X_509		= 0x00000000,
+	PKCS11_CKC_X_509_ATTR_CERT	= 0x00000001,
+	PKCS11_CKC_WTLS			= 0x00000002,
+	/* Vendor extension: reserved for undefined ID (~0U) */
+	PKCS11_CKC_UNDEFINED_ID		= PKCS11_UNDEFINED_ID,
+};
+
+/*
+ * Valid values for attribute PKCS11_CKA_CERTIFICATE_CATEGORY
+ */
+enum pkcs11_certificate_category {
+	PKCS11_CK_CERTIFICATE_CATEGORY_UNSPECIFIED	= 0UL,
+	PKCS11_CK_CERTIFICATE_CATEGORY_TOKEN_USER	= 1UL,
+	PKCS11_CK_CERTIFICATE_CATEGORY_AUTHORITY	= 2UL,
+	PKCS11_CK_CERTIFICATE_CATEGORY_OTHER_ENTITY	= 3UL,
```

Comment:
OK

Reply:
In pkcs11 TA,all IDs are 32bit value. Here U should be better than UL.
Note optee tends to use macros `U()` and `UL()` instead of prefix `U` or `UL`.